tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -731,5 +739,32 @@ void TrySetValue(int index)
 
             return httpErrorCodesArray;
         }
+
+        internal static DbmPropagationLevel ValidateDbmPropagationInput(string inputValue)
+        {
+            DbmPropagationLevel propagationValue;
+
+            inputValue = inputValue.ToLower();
+
+            if (inputValue.Equals("disabled", StringComparison.Ordinal))
+            {
+                propagationValue = DbmPropagationLevel.Disabled;
+            }
+            else if (inputValue.Equals("service", StringComparison.Ordinal))
+            {
+                propagationValue = DbmPropagationLevel.Service;
+            }
+            else if (inputValue.Equals("full", StringComparison.Ordinal))
+            {
+                propagationValue = DbmPropagationLevel.Full;
+            }
```

Comment:
Small nit: instead of using `ToLower()` (which may allocate a new string), you can use `StringComparison.OrdinalIgnoreCase`:
```suggestion
            if (inputValue.Equals("disabled", StringComparison.OrdinalIgnoreCase))
            {
                propagationValue = DbmPropagationLevel.Disabled;
            }
            else if (inputValue.Equals("service", StringComparison.OrdinalIgnoreCase))
            {
                propagationValue = DbmPropagationLevel.Service;
            }
            else if (inputValue.Equals("full", StringComparison.OrdinalIgnoreCase))
            {
                propagationValue = DbmPropagationLevel.Full;
            }
```